src/shapes/Object/BBox.ts

```diff
@@ -0,0 +1,313 @@
+import { iMatrix } from '../../constants';
+import { Point } from '../../Point';
+import { TCornerPoint, TMat2D, TOriginX, TOriginY } from '../../typedefs';
+import { mapValues } from '../../util/internals';
+import { makeBoundingBoxFromPoints } from '../../util/misc/boundingBoxFromPoints';
+import {
+  invertTransform,
+  multiplyTransformMatrices,
+} from '../../util/misc/matrix';
+import {
+  calcBaseChangeMatrix,
+  sendPointToPlane,
+} from '../../util/misc/planeChange';
+import { degreesToRadians } from '../../util/misc/radiansDegreesConversion';
+import { resolveOrigin } from '../../util/misc/resolveOrigin';
+import { createVector } from '../../util/misc/vectors';
+import type { ObjectGeometry } from './ObjectGeometry';
+
+const CENTER_ORIGIN = { x: 'center', y: 'center' } as const;
+
+export interface BBoxPlanes {
+  retina(): TMat2D;
+  viewport(): TMat2D;
+  parent(): TMat2D;
+  self(): TMat2D;
+}
+
+export type Coords = [Point, Point, Point, Point] & TCornerPoint;
+
+export type OriginDiff = { x: TOriginX; y: TOriginY };
+
+export type TRotatedBBox = ReturnType<typeof BBox['rotated']>;
+
+export class PlaneBBox {
+  private readonly originTransformation: TMat2D;
+
+  static build(coords: TCornerPoint) {
+    return new this(
+      calcBaseChangeMatrix(
+        undefined,
+        [
+          createVector(coords.tl, coords.tr),
+          createVector(coords.tl, coords.bl),
+        ],
+        coords.tl.midPointFrom(coords.br)
+      )
+    );
+  }
+
+  protected constructor(transform: TMat2D) {
+    this.originTransformation = Object.freeze([...transform]) as TMat2D;
+  }
+
+  getTransformation() {
+    return this.originTransformation;
+  }
+
+  getCoordMap() {
+    return {
+      tl: this.pointFromOrigin(new Point(-0.5, -0.5)),
+      tr: this.pointFromOrigin(new Point(0.5, -0.5)),
+      br: this.pointFromOrigin(new Point(0.5, 0.5)),
+      bl: this.pointFromOrigin(new Point(-0.5, 0.5)),
+    };
+  }
+
+  getCoords() {
+    const { tl, tr, br, bl } = this.getCoordMap();
+    return [tl, tr, br, bl];
+  }
```

Comment:
just a note or i forget. Is this `Object.values(this.getCoordMap())`?

Reply:
yes
I should redo that